File: 739.java
import java.util.*;
import javafx.util.*;

class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        Deque<Pair<Integer, Integer>> monoStack = new ArrayDeque<>();
        int n = temperatures.length;
        int[] result = new int[n];
        for (int i = 0; i < n; i++) {
            while (!monoStack.isEmpty() && temperatures[i] > monoStack.peek().getValue()) {
                Pair<Integer, Integer> prevDay = monoStack.pop();
                result[prevDay.getKey()] = i - prevDay.getKey();
            }
            monoStack.push(new Pair<Integer, Integer>(i, temperatures[i]));
        }
        return result;
    }
}

File: Python/739.py
from typing import List


class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        result = [0] * len(temperatures)
        stack = []  # (temeprature, index)
        for i, temp in enumerate(temperatures):
            while stack and temp > stack[-1][0]:
                pair = stack.pop()
                result[pair[1]] = i - pair[1]
            stack.append((temp, i))

        return result
